contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -1288,6 +1326,66 @@ validate_special_function(char *func_nsname, char *func_name, List* fargs, int n
 	return true;
 }
 
+/*
+ * Selects the best candidate function for HASHBYTES based on input types.
+ * Focuses on matching the second argument type, particularly for NVARCHAR.
+ * Returns a single best candidate from the sys schema, or NULL if none found.
+ */
+static FuncCandidateList
+tsql_func_select_candidate_for_hashbytes(List *names, List *fargs, int nargs, Oid *input_typeids, FuncCandidateList candidates)
+{
+
+	FuncCandidateList			current_candidate, best_candidate;
+	Oid 						expr_second_arg;
+	char					   *proc_nsname;
+	char					   *proc_name;
+	int							ncandidates;
+	Oid							sys_oid = get_namespace_oid("sys", false);
+	Oid 						*argtypes;
+	int							nargs_func = 0;
+	Oid							second_arg;
+
+	DeconstructQualifiedName(names, &proc_nsname, &proc_name);
+	
+	/* if common_utility_plugin_ptr is not initialised */
+	if (common_utility_plugin_ptr == NULL)
+		ereport(ERROR,
+				(errcode(ERRCODE_INTERNAL_ERROR),
+					errmsg("Failed to find common utility plugin.")));
+
+	expr_second_arg = InvalidOid;
+	if ((*common_utility_plugin_ptr->is_tsql_nvarchar_datatype)(input_typeids[1]))
+	{
+		expr_second_arg = input_typeids[1];		
+	}
+
+	/* Get the candidate with matching second argument type */
+	ncandidates = 0;
+	best_candidate = NULL;
+
+	for (current_candidate = candidates;
+		current_candidate != NULL;
+		current_candidate = current_candidate->next)
+	{
+		/* we should only consider candidates for hashbytes function from sys schema */
+		if (get_func_namespace(current_candidate->oid) != sys_oid)
+			return NULL;
+
+		get_func_signature(current_candidate->oid,&argtypes, &nargs_func);
+		second_arg = argtypes[1];
+		if(second_arg == expr_second_arg)
```

Comment:
can we just compare `argtypes[1] ==  input_typeids[1];`
and which ever function matches, we select it. For no match fall back to current behaviour.